app/controllers/api/base_controller/generic.rb

```diff
@@ -253,9 +251,10 @@ def model_ident(model, type = nil)
         end
       end
 
-      def validate_id(id, type, klass)
-        return if collection_config.resource_identifier(type) != "id"
-        raise NotFoundError, "Invalid #{klass} id #{id} specified" unless id.kind_of?(Integer) || id =~ /\A\d+\z/
+      def validate_id(id, key_id, klass)
+        if key_id == "id" && !id&.integer? || id.nil?
+          raise NotFoundError, "Invalid #{klass} #{key_id} #{id} specified"
```

Comment:
```suggestion
          raise NotFoundError, "Invalid #{klass} #{key_id} #{id.inspect} specified"
```

Reply:
I didn't like the double quotes coming across, I did an `|| 'nil'`